crowbar_framework/app/models/service_object.rb

```diff
@@ -1168,6 +1168,9 @@ def apply_role(role, inst, in_queue)
     # of names of Chef roles.
     run_order = []
 
+    # get databag to remember potential removal of a role
+    databag = ProposalObject.find_proposal(@bc_name, inst)
```

Comment:
no you are right, we need sth like
`old_nodes = old_elements[role_name] || nodes_from_to_remove`